content/source/docs/enterprise/api/ptfe_settings.html.md

```diff
@@ -0,0 +1,491 @@
+      "sso_api_token_session_timeout": 1209600
+    }
+  }
+}
+```
+
+## Update SAML Settings
+`PATCH /api/v2/admin/settings/saml`
+
+-> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/service-accounts.html#organization-service-accounts). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) which has the `site-admins` permission.
+
+Status  | Response                                         | Reason
+--------|--------------------------------------------------|-------
+[200][] | Array of [JSON API document][]s (`type: "settings"`) | Successfully updated SAML settings
+[404][] | [JSON API error object][]                    | User unauthorized to perform action
+[422][] | [JSON API error object][]                    | Malformed request body (missing attributes, wrong types, etc.)
+
+[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
+[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+[422]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
+[JSON API document]: https://www.terraform.io/docs/enterprise/api/index.html#json-api-documents
+[JSON API error object]: http://jsonapi.org/format/#error-objects
+
+### Request Body
+
+This PATCH endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required if `data.attributes.enabled` is set to `true`.
+
+Key path                    | Type   | Default | Description
+----------------------------|--------|---------|------------
+`data.attributes.enabled`    | bool   | `false` | Allows SAML to be used. If true, all remaining attributes are required.
+`data.attributes.idp_cert`   | string |         | Identity Provider Certificate specifies the PEM encoded X.509 Certificate as provided by the IdP configuration.
+`data.attributes.slo_target` | string |         | Single Log Out URL specifies the HTTP(s) endpoint on your IdP for single logout requests. This value is provided by your IdP configuration.
+`data.attributes.sso_target` | string |         | Single Sign On URL specifies the HTTP(S) endpoint on your IdP for single sign-on requests. This value is provided by your IdP configuration.
+`data.attributes.attr_groups`| string | `"MemberOf"` | Team Attribute Name specifies the name of the SAML attribute that determines team membership. The value of this attribute in the SAML assertion must be a string containing a comma-separated list of team names.
+`data.attributes.attr_site_admin`| string |`"SiteAdmin"`| Specifies the role for site admin access, overridding the "Site Admin Role" method.
+`data.attributes.site_admin_role`| string |`"site-admins"`| Specifies the role for site admin access, provided in the list of roles sent in the Team Attribute Name attribute.
+`data.attributes.sso_api_token_session_timeout`| integer | 1209600 | Specifies the Single Sign On session timeout, defaulting to 14 days.
```

Comment:
Are there maximum/minimum values for this attribute? What happens if it's elided? Can session timeout be disabled?

Reply:
I'm unsure. There appears to be no current validation on this value. @zhssh are you aware of a min/max for session timeout and if it can be nil?